### PR TITLE
fix(opensearch): correctly validate ebs configuration against instance types 

### LINKED ISSUE #11898: (elasticsearch) instance type and ebs validation doesn't distinguish data and master node during domain creation.
When trying to create Elasticsearch domain with master node type `c5.2xlarge.elasticsearch`, data node type `i3.2xlarge.elasticsearch`, with ebs disabled, cdk throws exception says
```
throw new Error('EBS volumes are required when using instance types other than r3 or i3.');
```
If I enable ebs, it throws exception:
```
throw new Error('I3 instance types do not support EBS storage volumes.');
```
The version I am using is 1.76.

From API reference, the ebs suppose to be only attached to master node in my case, however the validation couldn't distinguish data node and master node. With my instance config, it thinks c5 should have ebs, but i3 shouldn't.

c5 is recommended for master node, and i3 for data node. With this bug, there is no way to use CDK to create Elasticsearch domain with master nodes.

### PR DESCRIPTION
**Pull request by nom3ad**
Currently, the following valid cluster configurations are not possible to be created with CDK.

```ts
new es.Domain(stack, 'Domain1', {
    version: es.EngineVersion.OPENSEARCH_1_0,
    ebs: { enabled: false },
    capacity: {
        masterNodes: 3,
        masterNodeInstanceType: 'c5.2xlarge.search',
        dataNodeInstanceType: 'i3.2xlarge.search'
    }
});


new es.Domain(stack, 'Domain2', {
    version: es.EngineVersion.OPENSEARCH_1_0,
    ebs: { enabled: false, },
    capacity: {
        masterNodes: 3,
        masterNodeInstanceType: 'c6g.large.search',
        dataNodeInstanceType: 'r6gd.large.search'
    }
});
```
Throws error:
`EBS volumes are required when using instance types other than r3, i3 or r6gd.`

Here, EBS validation is being checked on master nodes instead of just the data nodes -  which causes the above failure.

This PR applies fix on both `elasticsearch` and `opensearchservice` modules

Fixes #11898 

----
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nom3ad** [comment]
@kaizen3031593 Would you mind taking a look when you get time?

**Review by kaizen3031593 — COMMENTED** [review]
Hi @nom3ad, sorry for the delay! Thanks for the PR and double thanks for providing the fix for both `elasticsearch` and `opensearch`.

I read through the linked issue, and it seems like folks are/were [reporting](https://github.com/aws/aws-cdk/issues/11898#issuecomment-893479782) a Cloudformation validation even when using the CDK escape hatch. Given that's the case, I'm not convinced that doing anything on the CDK side will help alleviate those issues -- and I'm curious to hear how you've gotten past that!

My gut tells me that even if we go ahead and merge this PR, the Cloudformation validation will still be there. Please let me know if that's not the case!

**Comment by nom3ad** [comment]
@kaizen3031593 Thanks for looking in to it.

> My gut tells me that even if we go ahead and merge this PR, the Cloudformation validation will still be there. Please let me know if that's not the case!

Hmm,  I don't think CloudFormation is throwing a false positive validation error in this scenario.  I have successfully able to work around  CDK validation by https://github.com/aws/aws-cdk/issues/11898#issuecomment-940087117  and CloudFormation happily deployed it. 
Likewise, I can confirm the examples provided in the PR description https://github.com/aws/aws-cdk/pull/16911#issue-1022930373 is indeed deployable as I actually tried them while working on the fix.  :-) 

> I read through the linked issue, and it seems like folks are/were reporting a Cloudformation validation even when using the CDK escape hatch. Given that's the case, I'm not convinced that doing anything on the CDK side will help alleviate those issues -- and I'm curious to hear how you've gotten past that!

I have looked into the [reported code snippet](https://github.com/aws/aws-cdk/issues/11898#issuecomment-893479782). It should generate a template as follows (removed some unrelated properties):
```yaml
Resources:
  ElasticsearchCluster167EBEDC:
    Type: AWS::Elasticsearch::Domain
    Properties:
      EBSOptions:
        EBSEnabled: true   # data node instance type r5.large.elasticsearch works with EBS enabled but i3.large.elasticsearch does not.
        VolumeSize: 10           
        VolumeType: gp2
      ElasticsearchClusterConfig:
        DedicatedMasterCount: 3
        DedicatedMasterEnabled: true
        DedicatedMasterType: m5.xlarge.elasticsearch  # master instance type doesn't care about EBS configuration
        InstanceCount: 4
        InstanceType: r5.large.elasticsearch    # this is the default data node instance type when users don't provide one.
        ZoneAwarenessEnabled: false
      ElasticsearchVersion: "7.7"
```
And the above template is deployable too.  But the reported error was:

> Instance type i3.large.elasticsearch does not support EBS storage (Service: AWSElasticsearch; Status Code: 400; Error Code: ValidationException; Request ID: 0dad9072-8baa-4390-a519-0c3233656a56; Proxy: null)

As we can see `i3.large.elasticsearch` in the logs,  it's clear that, provided snippet was not deployed intact.  I guess the user patched  `InstanceType: r5.large.elasticsearch` to `i3.large.elasticsearch` -  which actually results in an invalid template -  because [i3.large does not support EBS](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html).    
So nothing to blame on CloudFormation in this particular case.

Hope this clears the concern.

**Comment by nom3ad** [comment]
@kaizen3031593 would you mind taking a look again?

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts [hunk 1]
```diff
@@ -1446,6 +1456,40 @@ describe('custom error responses', () => {
...
+      ebs: {
+        enabled: false,
+      },
+      capacity: {
+        masterNodes: 3,
+        masterNodeInstanceType: 'c6g.large.elasticsearch',
+        dataNodeInstanceType: 'r6gd.large.elasticsearch',
+      },
+    });
+    expect(stack).toCountResources('AWS::Elasticsearch::Domain', 2);
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:1492]:
```suggestion
    // both configurations pass synth-time validation
    expect(stack).toCountResources('AWS::Elasticsearch::Domain', 2);
```

File: packages/@aws-cdk/aws-opensearchservice/test/domain.test.ts [hunk 2]
```diff
@@ -1448,6 +1458,40 @@ describe('custom error responses', () => {
...
+      ebs: {
+        enabled: false,
+      },
+      capacity: {
+        masterNodes: 3,
+        masterNodeInstanceType: 'c6g.large.search',
+        dataNodeInstanceType: 'r6gd.large.search',
+      },
+    });
+    expect(stack).toCountResources('AWS::OpenSearchService::Domain', 2);
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-opensearchservice/test/domain.test.ts:1494]:
```suggestion
    // both configurations pass synth-time validation
    expect(stack).toCountResources('AWS::OpenSearchService::Domain', 2);
```

**Review by kaizen3031593 — APPROVED** [review]
Sorry for the delay (again). Thanks for the detailed reply, this does look like an excess CDK validation. Just going to add a comment in the tests to explain why we are happy to just assert that the domains exist in the template.

**Review by kaizen3031593 — APPROVED** [review]

**Pull request merged**